src/lib/Instance.ts

```diff
@@ -0,0 +1,33 @@
+export default class Instance {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _onChanged: ((...args: any[]) => void)[];
+  onChanged: {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    connect: (callback: (...args: any[]) => void) => () => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fire: (...args: any[]) => void;
+  };
+
+  constructor() {
+    this._onChanged = [];
+    this.onChanged = {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      connect: (callback: (...args: any[]) => void) => {
+        let active = true;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const wrappedCallback = (...args: any[]) =>
+          active ? callback(...args) : undefined;
+
+        this._onChanged.push(wrappedCallback);
+
+        return () => (active = false);
+      },
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      fire: (...args: any[]) => {
+        this._onChanged.forEach((callback) =>
+          setTimeout(() => callback(...args), 0)
+        );
+      },
```

Comment:
```suggestion
      connect: (callback: (...args: unknown[]) => void) => {
        let active = true;
        const wrappedCallback = (...args: unknown[]) =>
          active ? callback(...args) : undefined;

        this._onChanged.push(wrappedCallback);

        return () => (active = false);
      },
      fire: (...args: unknown[]) => {
        this._onChanged.forEach((callback) =>
          setTimeout(() => callback(...args), 0)
        );
      },
```
Assuming you do not want to use generics & a separate class, use unknown[] - I don't see the point in any[] as unknown[] works just fine here